THE-app/app/(tabs)/booksScreen.tsx

```diff
@@ -30,15 +31,105 @@ const GET_BOOKS = gql`
 
 function BooksScreen() {
   const { loading, error, data } = useQuery(GET_BOOKS);
+  const [filterText, setFilterText] = useState('');
+  const [selectedSubject, setSelectedSubject] = useState('');
+  const [selectedLevel, setSelectedLevel] = useState('');
+
+  let subjectOptions: JSX.Element[] = [];
+  let levelOptions: JSX.Element[] = [];
+
+  if (!loading && !error && data) {
```

Comment:
If you moved the loading and error checks above this (called early return), you wouldn't need to check.

Either way, in loading and error cases, you wouldn't have `data`, so you probably don't need to check all three.